src/rewrite/strategy.rs

```diff
@@ -53,6 +53,50 @@ pub trait RewriteStrategy {
     }
 }
 
+/// The result of a rewrite strategy.
+///
+/// Returned by [`RewriteStrategy::apply_rewrites`].
+pub struct RewriteResult<Cost: CircuitCost> {
+    /// The rewritten circuits.
+    pub circs: Vec<Hugr>,
+    /// The cost delta of each rewritten circuit.
+    pub cost_deltas: Vec<Cost::CostDelta>,
+}
```

Comment:
What's the advantage of this over `type RewriteResult<Cost: CircuitCost> = Vec<(Hugr, Cost::CostDelta)>`?

You'd get all of the methods below for free. Otherwise keep it this way and use `delegate!`.

Reply:
I guess it makes testing a bit easier than having to play with indices.

Reply:
Yeah, specially since the fields are public it's nicer if you can name them.